# Bacterial_genomics_software_tools
A **very** non-exhaustive curated list of software tools for various aspects of prokaryote genomic analyses - mostly command line tools.  For online options, please see: https://github.com/zadyson/Online-genomics-software-tools.  Please note I am not affiliated with these tools, unless otherwise stated, nor have I explored their full functionality.

## Gene screening (e.g. AMR, virulence, plasmids)
SRST2: https://github.com/katholt/srst2  
ARIBA: https://github.com/sanger-pathogens/ariba  
AbritAMR ISO acredited AMR detection pipeline for public health labs: https://github.com/MDU-PHL/abritamr  

## Plasmids
MOB-suite: https://github.com/phac-nml/mob-suite 
PLSDB: https://ccb-microbe.cs.uni-saarland.de/plsdb  

## Prophages
VIR-sorter: https://github.com/simroux/VirSorter  
VIR-sorter2: https://github.com/jiarong/VirSorter2  
ViPTreeGen: https://github.com/yosuken/ViPTreeGen  

## Insertion Sequneces
ISMapper: https://github.com/jhawkey/IS_mapper  

## CRISPR-Cas
CRISPR Comparison ToolKit (CCTK): https://github.com/Alan-Collins/CRISPR_comparison_toolkit   
CRISPRCasFinder: https://crisprcas.i2bc.paris-saclay.fr/Home/Download  
CRISPR Visualiser: github.com/CRISPRlab/CRISPRviz  
CRISPRdisco: http://github.com/crisprlab/CRISPRdisco
CRISPR-Detector: https://github.com/hlcas/CRISPRdetector  
CRIPSR Recognition Tool (CRT): http://www.room220.com/crt  
PILER-CR: http://www.drive5.com/pilercr  

## Mapping and variant calling
snippy; https://github.com/tseemann/snippy  
RedDog: https://github.com/katholt/RedDog  
ParSNP: https://harvest.readthedocs.io/en/latest/content/parsnp.html  

## Phylogenetic inference:
RAxML: https://github.com/stamatak/standard-RAxML  
FastTree: http://www.microbesonline.org/fasttree/  
iqtree: http://www.iqtree.org/  

## Phylodynamics
BEAST: https://beast.community/  
BEAST2: https://www.beast2.org/  
TreeDater: https://github.com/emvolz/treedater  
BactDating: https://github.com/xavierdidelot/BactDating  
Skygrowth: https://github.com/mrc-ide/skygrowth  
Treemer: https://git.scicore.unibas.ch/TBRU/Treemmer  
TipDatingBEAST: https://github.com/cran/TipDatingBeast  

## Clustering
PopPUNK: https://poppunk.net/  
HierBAPS: https://github.com/gtonkinhill/rhierbaps  
FastBAPS: https://github.com/gtonkinhill/fastbaps   

# LD  
Lian: http://guanine.evolbio.mpg.de/cgi-bin/lian/lian.cgi.pl   
Ia measure via Poppr: https://grunwaldlab.github.io/poppr/reference/poppr-package.html

## Haplotype networks  
popART: https://github.com/jessicawleigh/popart.  

## Typing
MLST: https://github.com/tseemann/mlst   
Mykrobe: https://github.com/Mykrobe-tools/mykrobe  

## Transmission chain reconstruction
Transphylo: https://github.com/xavierdidelot/TransPhylo  
Outbreaker2: https://github.com/reconhub/outbreaker2  

## Assembly
Unicycler: https://github.com/rrwick/Unicycler  
SPAdes: https://github.com/ablab/spades  

## Annotation
PROKKA: https://github.com/tseemann/prokka  
Bakta: https://github.com/oschwengers/bakta  

## Pangenome
Roary: https://sanger-pathogens.github.io/Roary/  
Panstripe: https://github.com/gtonkinhill/panstripe/  

## Microbiome (16S)
QIIME2: https://qiime2.org/  
DESeq2: https://joey711.github.io/phyloseq-extensions/DESeq2.html  

## Read and assembly manipulation
Fastaq: https://github.com/sanger-pathogens/Fastaq  

## PCR primer design and validation  
RUCS: https://bitbucket.org/genomicepidemiology/rucs/src/master/  
PrimerBLAST: https://www.ncbi.nlm.nih.gov/tools/primer-blast/  
MFEPrimer: https://mfeprimer3.igenetech.com/specificity 

## In silico PCR
https://anaconda.org/bioconda/ispcr

## Machine learning from WGS  
GenoLearn: https://genolearn.readthedocs.io/en/latest/  

## Growth curve analysis: 
Growthcurver: https://github.com/sprouffske/growthcurver  

## Distances
PyANI: https://github.com/widdowquinn/pyani  
